tests/rptest/services/kgo_verifier_services.py

```diff
@@ -442,14 +444,24 @@ def _ingest_status(self, worker_statuses):
         self._parent._status = reduced
 
     def poll_status(self):
+        retry_strategy = Retry(total=5,
+                               connect=5,
+                               read=5,
+                               backoff_factor=1,
+                               status=5,
+                               allowed_methods=['GET'],
+                               status_forcelist=[503, 504])
+        session = requests.Session()
+        session.mount("http://", HTTPAdapter(max_retries=retry_strategy))
+
         while not self._stop_requested.is_set():
             drop_out = self._shutdown_requested.is_set()
-
-            r = requests.get(self._parent._remote_url(self._node, "status"),
-                             timeout=5)
+            r = session.get(url=self._parent._remote_url(self._node, "status"),
+                            timeout=5)
             r.raise_for_status()
             worker_statuses = r.json()
-            self._ingest_status(worker_statuses)
+            if worker_statuses != None:
```

Comment:
What prompted this change?

Reply:
The only way this can return None from what I'm reading is if the server responded with json content "null" 🤔 

Otherwise, it should have raised in raise_for_status or during get request 🤔

Reply:
the None is returned on connection error

Reply:
> the None is returned on connection error

requests.get throws on connection error. Do you have a specific type of error in mind?

I tried `requests.get('127.0.0.1:42')` without any listeners and it throws from around there https://github.com/urllib3/urllib3/blob/e4d253e8af19e30acfc1df5dab19803f60e57225/src/urllib3/connection.py#L192-L219